apps/examples/binary_update/binary_update_main.c

```diff
@@ -78,20 +110,130 @@ static void binary_update_getinfo_all(void)
 	}
 }
 
-static void binary_update_getinfo(char *name)
+static void binary_update_info(binary_info_t *pre_bin_info, binary_info_t *cur_bin_info, int condition)
+{
+	printf(" === [%4s] Update info === \n", cur_bin_info->name);
+	printf(" %4s | %8s | %s \n", "Con", "Version", "Inactive part");
+	printf(" ------------------------- \n");
+	printf(" %4s | %8s | %s \n", "Pre", pre_bin_info->version, pre_bin_info->dev_path);
+	printf(" %4s | %8s | %s \n", "Cur", cur_bin_info->version, cur_bin_info->dev_path);
+	printf(" ========================= \n");
+
+	if (condition == DOWNLOAD_SUCCESS) {
+		if (strncmp(pre_bin_info->dev_path, cur_bin_info->dev_path, sizeof(pre_bin_info->dev_path) != 0)) {
+			fail_cnt++;
+			printf("Binary Update Download Success Case FAIL.\n");
+		} else {
+			printf("Binary Update Download Success Case SUCCESS.\n");
+		}
+	} else { //DOWNLOAD_FAIL
+		if (strncmp(pre_bin_info->dev_path, cur_bin_info->dev_path, sizeof(pre_bin_info->dev_path) == 0)) {
+			fail_cnt++;
+			printf("Binary Update Download Fail Case FAIL.\n");
+		} else {
+			printf("Binary Update Download Fail Case SUCCESS.\n");
```

Comment:
Results should be 4 cases like below
1. success to load valid higher version binary
2. fail to load valid higher version binary
3. no update with invalid binary
4. warning!! load invalid binary